app/assets/javascripts/views/tasks/collection_objects/area_and_date/index.js

```diff
@@ -0,0 +1,255 @@
+      //var area_selector = $("#geographic_area_id_for_by_area");
+  
+      $(".result_map_toggle").click(function (event) {           // switch to the map view
+        $("#area_count").text('????');
+        $("#show_list").attr('hidden', true);         // hide the list view
+        $("#show_map").removeAttr('hidden');          // reveal the map
+        $(".result_list_toggle").removeAttr('hidden');           // expose the other link
+        $(".result_map_toggle").attr('hidden', true);
+        $("[name='[geographic_area_id]']").attr('value', '');
+        result_map = _init_simple_map();
+        result_map = TW.vendor.lib.google.maps.initializeMap('simple_map_canvas', result_collection);
+      });
+  
+      $(".result_list_toggle").click(function (event) {          // switch to the list view
+        $("#area_count").text('????');
+        $("#show_map").attr('hidden', true);          // hide the map
+        $("#show_list").removeAttr('hidden');         // reveal the area selector
+        $(".result_map_toggle").removeAttr('hidden');            // expose the other link
+        $(".result_list_toggle").attr('hidden', true);
+        $("#drawn_area_shape").attr('value', '');
+      });
+  
+      $("#set_taxon_name").click(function (event) {
+          $("taxon_count").text('????');
+        $("select_taxon_name").mx_spinner('show');
+    
+        // do the GET here.
+        $("select_taxon_name").mx_spinner('hide');
+          event.preventDefault();
+        }
+      );
+      
+      $("#set_area").click(function (event) {      // register the click handler for the made-from-scratch-button
+        $("#area_count").text('????');
+          //var geo_id = $("input[name='[geographic_area_id]']").val();
+        $("#select_area").mx_spinner('show');
+        $.get('set_area', $("#set_area_form").serialize(), function (local_data) {
+              var popcorn = local_data;
+              $("#area_count").text(local_data.html);
+            $("#select_area").mx_spinner('hide');
+          }, 'json'  // I expect a json response
+          );
+  
+        event.preventDefault();
+        }
+      );
+      // below could be used as template for auto-get on .overlaycomplete
+      //$("#set_area").on("ajax:success", function (e, data) {
+      //    $("#area_count").text(data.html);
+      //    return true;
+      //  }
+      //).on("#ajax:errror", function (e, xhr, status, error) {
+      //    $("area_count").text("<p>set_area error => " + error + "</p>")
+      //  }
+      //);
+      $("#find_area_and_date_commit").click(function (event, href) {
+        //$("#result_span").text('**********');
+        if (href == undefined) {
+          href = $("#set_area_form").serialize() + '&' + $("#set_date_form").serialize() + '&' + $("#set_taxon_name_form");
+        }
+        $("#find_item").mx_spinner('show');
+        $.get('find', href, function (local_data) {
+          //var html = local_data.html;
```

Comment:
These can be removed, but should probably be retained elsewhere, as this is basically a template for the ajax exchange.